crates/pipeline-manager/src/compiler/rust_compiler.rs

```diff
@@ -489,8 +497,14 @@ pub async fn perform_rust_compilation(
             &source_checksum,
             &profile,
         )
-        .await?;
-    }
+        .await?
+    } else {
+        RustCompilationInfo {
```

Comment:
so the not cached case always returns success?

Reply:
The cached case always returns success

Reply:
The not cached case can return an Err, which gets caught by the ?